### PR TITLE
raise exception when attempting to install from this branch

### PR DESCRIPTION
**Pull request by CNDW**
context:
the pipenv install process we use runs `pipenv lock --keep-outdated -r > ./requirements.txt` and then a `pip install -r requirements.txt`. Currently we are installing Flask-HAL from a branch on this fork with something like 
```
Flask-HAL = {git = "https://github.com/GainCompliance/Flask-HAL.git",ref = "self_attr"}
```
in the `Pipfile`

I have occasionally seen local installs fail with an error during the `pip install -r requirements.txt` with this message in the terminal
```
ERROR: Invalid requirement: "Switched to a new branch 'self_attr'" (from line 1 of ./requirements.txt)
```
This comes from `pipenv lock > ./requirements.txt` not properly filtering the stdout during the git process of downloading the repo and switching to the branch when installing.

In the past this issue has been intermittent and subsequent installs don't seem to have the issue. During the work on converting document-api to python3, this issue started to present in CI for unknown reasons and was completely blocking install. To work around it, I created a new repo https://github.com/GainCompliance/pygain-flask-hal based on this branch, gutted the CI/development config and rebuilt it to publish on gemfury. 

Merging this PR will make all builds fail when they attempt to install from the `self_attr` branch with a message to update dependencies in `Pipfile`. Is this something we want to do? Should we do it now and just force update, or should I manually update all of the repos first?

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -9,61 +9,56 @@
 specification for your ``REST`` Flask Applications.
 """
 
+replacement_req = 'pygain-flask-hal = {version = "==1.0.6",index = "fury"}'
+raise Exception(
+    "https://github.com/GainCompliance/Flask-HAL.git@self_attr is depricated, please change your dependency to {}".format(
```
Comment (jcjones1515) [inline setup.py]: deprecated

**Review by jcjones1515 — APPROVED** [review]
makes sense to me

**Comment by davidan1981** [comment]
Do you know which repos need to be converted?

**Comment by CNDW** [comment]
the repos that need to be updated are 
```
auth-api
filing-api
data-api
document-render-api
meta-api
document-api
validation-api
gain-python-util (probably a deprecated lib?)
texxt-block-api
meta-api
```

so yea.... pretty much all of them. taking this approach would just mean that individuals will update those repos when they are touched

**Pull request merged**